src/connection-manager/index.ts

```diff
@@ -152,6 +164,8 @@ export class DefaultConnectionManager extends EventEmitter<ConnectionManagerEven
   private readonly startupReconnectTimeout: number
   private connectOnStartupController?: TimeoutController
   private readonly dialTimeout: number
+  private readonly allow: Multiaddr[]
```

Comment:
Would it make sense to use a `Set` here?

Reply:
I don't think so because we call `.some` on it in order to see if the connecting peer's multiaddr is prefixed by anything in the allow list - to do that with a `Set` we'd have to convert it to an array first.

We could have a separate function that iterates over the Set with a `for..of` doing the check that returns early, but `.some` is a one-liner that does the same thing so there seems little benefit?